packages/expo-router/src/doctor/index.ts

```diff
@@ -0,0 +1,90 @@
+import createDebug from 'debug';
+import type { PackageJSONConfig } from 'expo/config';
+import semverRangeSubset from 'semver/ranges/subset';
+
+const debug = createDebug('expo:router:doctor');
+
+/**
+ * Small hack to get the package.json.
+ * We do no use import() as this would require changing the rootDir in `tsconfig.json`,
+ * which in turn will change the structure of the outDir.
+ */
+const routerPkg: PackageJSONConfig = require('../../package.json');
+
+const routerDependencies = Object.entries(
+  Object.assign({}, routerPkg.dependencies, routerPkg.peerDependencies)
+).filter((entry) => entry[1] !== '*');
+
+type IncorrectDependency = {
+  packageName: string;
+  packageType: 'dependencies' | 'devDependencies';
+  expectedVersionOrRange: string;
+  actualVersion: string;
+};
+
+export function doctor(
+  pkg: PackageJSONConfig,
+  appReactNavigationPath: string
+): IncorrectDependency[] {
+  const resolvedDependencies = { ...pkg.dependencies, ...pkg.devDependencies };
+  const libReactNavigationPath = require.resolve('@react-navigation/native');
+
+  const userExcluded = new Set(pkg.expo?.install?.exclude);
+  const incorrectDependencies: IncorrectDependency[] = [];
+
+  /**
+   * If the user has a dependency with a sub-dependency on @react-navigation/native, this may install a different
+   * version of @react-navigation/native than the one required by expo-router.
+   *
+   * To detect this, we require the caller of this function to first resolve their path to @react-navigation/native, as
+   * they will get the 'top' level package. When expo-router resolves the path to @react-navigation/native, if it is different
+   * when the versions must not have matched and the package manager installed a nested node_module folder with a different
+   * version of @react-navigation/native.
+   */
+  if (
+    userExcluded.has('@react-navigation/native') &&
+    appReactNavigationPath !== libReactNavigationPath
+  ) {
+    console.warn(
+      `There are two different versions of @react-navigation/native in your 'node_modules', this may lead to unexpected navigation behavior. Please ensure all dependencies that require '@react-navigation/native' depend on a version that matches the version range ${routerPkg.dependencies!['@react-navigation/native']}. You can silence this warning by adding '@react-navigation/native' to the 'expo.install.exclude' field in your package.json.`
```

Comment:
```suggestion
      `Detected multiple versions of ${chalk.bold('@react-navigation/native')} in your ${chalk.bold('node_modules')}. This may lead to unexpected navigation behavior and errors. ${learnMore('https://expo.fyi/router-navigation-deps')}.`
```

let's send people to an fyi page that explains how to fix and silence the warnings rather than printing a bunch of text here. this makes it easier for us to properly explain what is happening and how to fix, and to update it at any time to  add further clarification if needed.

Reply:
(i know chalk isn't imported here and we don't have a learnMore function as we do in other clis, so we'll need to add those here if we want to have nice formatting like the above)

also, it's best to avoid saying please in our messaging. if every message we print says 'please', we risk becoming polite to the point of annoyance, like the british (jk love you guys).